shell/platform/android/flutter_main.cc

```diff
@@ -108,14 +108,19 @@ void ConfigureShorebird(std::string android_cache_path,
   app_parameters.release_version = version.c_str();
   app_parameters.cache_dir = cache_dir.c_str();
 
-  // This seems to be ['libapp.so', 'real/path/to/libapp.so']
-  // not sure why, but we need to use the real path (second entry).
-  app_parameters.original_libapp_path =
-      settings.application_library_path.back().c_str();
+  std::vector<const char*> original_libapp_paths;
+  for (auto& path : settings.application_library_path) {
+    original_libapp_paths.push_back(strdup(path.c_str()));
```

Comment:
strdup does a malloc, so you need to do a free() to tear these down.

loop over original_libapp_paths, free() them and zero them out.

Reply:
You could also use vector<char>? https://www.techiedelight.com/convert-string-vector-chars-cpp/ and then vector will do the free for you.

Reply:
It depends on the contract of your call.  But typically a contract for a call like this would be that the caller can free the memory immediately after the call.  See how we do this when calling into this from Dart, we make a c string from Dart and then pass it in and then free it immediately.